draft-ietf-httpbis-message-signatures.md

```diff
@@ -1283,22 +1291,122 @@ The table below contains the initial contents of the HTTP Signature Specialty Co
-|`@request-response`|Active | {{content-request-response}} of this document|
+|`@request-response`|Active | Related-Response | {{content-request-response}} of this document|
 {: title="Initial contents of the HTTP Signature Specialty Component Identifiers Registry." }
 
 # Security Considerations {#security}
 
-(( TODO: need to dive deeper on this section; not sure how much of what's referenced below is actually applicable, or if it covers everything we need to worry about. ))
+## Use TLS {#security-tls}
+
+The use of HTTP Message Signatures does not negate the need for TLS or its equivalent to protect information in transit. Message signatures provide message integrity over the covered message components but do not provide any confidentiality for the communication between parties.
+
+TLS also protects the signature data itself from being captured by an attacker, which is an important step in preventing [signature replay](#security-replay).
+
+## Signature Replay {#security-replay}
+
+Since HTTP Message Signatures allows sub-portions of the HTTP message to be signed, it is possible for two different HTTP messages to validate against the same signature. The most extreme form of this would be a signature over no message components. If such a signature were intercepted, it could be replayed at will by an attacker, attached to any HTTP message. Even with sufficient component coverage, a given signature could be applied to two similar HTTP messages, allowing a message to be replayed by an attacker with the signature intact.
+
+To counteract these kinds of attacks, it's first important for the signer to cover sufficient portions of the message to differentiate it from other messages. In addition, the signature can use the `nonce` signature parameter to provide a per-message unique value to allow the verifier to detect replay of the signature itself if a nonce value is repeated. Furthermore, the signer can provide a timestamp for when the signature was created and a time at which the signer considers the signature to be invalid, limiting the utility of a captured signature value.
+
+If a verifier wants to trigger a new signature from a signer, it can send the `Accept-Signature` header field with a new `nonce` parameter. An attacker that is simply replaying a signature would not be able to generate a new signature with the chosen nonce value.
+
+## Insufficient Coverage {#security-coverage}
+
+Any portions of the message not covered by the signature are susceptible to modification by an attacker without affecting the signature. An attacker can take advantage of this by introducing a header field or other message component that will change the processing of the message but will not be covered by the signature. Such an altered message would still pass signature verification, but when the verifier processes the message as a whole, the unsigned content injected by the attacker would subvert the trust conveyed by the valid signature.
```

Comment:
I think "the verifier processes the message as a whole" conflates two components of the system: the piece that verifies the signature, and the piece that then processes the message. Could you introduce separate terms for those?

Reply:
By "verifier" we mean "the party that verifies the message with its signature" and not "the verification function". The function returns a boolean with the message as input, so the caller of that function would have access to the whole message anyway. I think having more precise terms isn't a bad idea, as long as we're using them consistently.